packages/fakta-ny-inntekt/src/components/tilkommetAktivitet/TilkommetAktivitet.tsx

```diff
@@ -0,0 +1,289 @@
+    dayjs(periode2.tom).isBetween(periode1.fom, periode1.tom, 'day', '[]');
+  return periode1OverlapperPeriode2 || periode2OverlapperPeriode1;
+};
+
+const andelFieldFinnesIPeriode = (
+  andelField: TilkommetInntektsforholdFieldValues,
+  periode: VurderInntektsforholdPeriode,
+): boolean =>
+  periode.inntektsforholdListe.some(
+    andel =>
+      andel.aktivitetStatus === andelField.aktivitetStatus &&
+      andel.arbeidsforholdId === andelField.arbeidsforholdId &&
+      andel.arbeidsgiverId === andelField.arbeidsgiverIdent,
+  );
+
+export const transformFieldValues = (
+  values: TilkommetAktivitetFieldValues,
+  bg: Beregningsgrunnlag,
+): BeregningsgrunnlagTilBekreftelse<VurderNyttInntektsforholTransformedValues> => {
+  const perioderFields = values.perioder;
+  const vurderInntektsforholdPerioder =
+    bg.faktaOmFordeling?.vurderNyttInntektsforholdDto?.vurderInntektsforholdPerioder || [];
+  const allePerioder = vurderInntektsforholdPerioder.flatMap(periode => {
+    const overlappendeFields = perioderFields.filter(p => overlapper(p, periode));
+    return overlappendeFields.map(periodeField => {
+      const andelFields = periodeField.inntektsforhold;
+      const transformerteInntektsforhold = andelFields
+        .filter(andelField => andelFieldFinnesIPeriode(andelField, periode))
+        .map(andelField => {
+          const skalUtbetalingReduseres = !!andelField.skalRedusereUtbetaling;
+          const bruttoInntektPrÅr = skalUtbetalingReduseres
+            ? removeSpacesFromNumber(andelField.bruttoInntektPrÅr)
+            : undefined;
+          return {
+            aktivitetStatus: andelField.aktivitetStatus,
+            arbeidsgiverId: andelField.arbeidsgiverIdent,
+            arbeidsforholdId: andelField.arbeidsforholdId,
+            skalRedusereUtbetaling: skalUtbetalingReduseres,
+            bruttoInntektPrÅr,
+          };
+        });
+      return {
+        fom: dayjs(periodeField.fom).isBefore(dayjs(periode.fom)) ? periode.fom : periodeField.fom,
+        tom: dayjs(periodeField.tom).isAfter(dayjs(periode.tom)) ? periode.tom : periodeField.tom,
+        tilkomneInntektsforhold: transformerteInntektsforhold,
+      };
+    });
+  });
+  return {
+    periode: values.periode,
+    begrunnelse: values.begrunnelse,
+    tilkomneInntektsforhold: allePerioder,
+  };
+};
+
+const transformValues = (
+  values: TilkommetAktivitetFormValues,
+  beregninsgrunnlagListe: Beregningsgrunnlag[],
+  vilkarperioder: Vilkårperiode[],
+): VurderNyttInntektsforholdAP => {
+  const fields = values[FORM_NAME];
+  const grunnlag =
+    fields
+      ?.filter(f => vurderesIBehandlingen(vilkarperioder, f.periode.fom))
+      .map(field => transformFieldValues(field, finnBeregningsgrunnlag(field.periode.fom, beregninsgrunnlagListe))) ??
+    [];
+  const begrunnelse = grunnlag?.map(gr => gr.begrunnelse).reduce((b1, b2) => (b1 !== null ? `${b1} ${b2}` : b2));
```

Comment:
Mangler initial verdi for reduce. Jeg ble usikker på om null eller undefined kunne skape problemer i en template string som her? 

```
const begrunnelse = grunnlag?.map(gr => gr.begrunnelse).reduce((b1, b2) => (b1 !== null ? `${b1} ${b2}` : b2), '');
```